### PR TITLE
Plugins: Update how translations are handled in the action modal

### PR DESCRIPTION
**Pull request by mattgawarecki**
This PR replaces the `getPluginActionDailog` method with a new hook that responds to changes in the user's locale via `useTranslate`. It also rewrites some plugin management-related code to reduce complexity and improve readability. All existing functionality should be preserved.

Resolves `1202619025189113-as-1202978931490015`.
Relates to #67799.

## Proposed Changes

* Create a new `useShowPluginActionDialog` hook (with supporting hook `useGetDialogText`).
* Create a higher-order component `withShowPluginActionDialog`, for a couple areas dependent on this functionality that don't (yet) support hooks.
* Add unit test coverage for both new hooks.
* Remove unused/unreferenced methods in the `PluginsList` component, and re-order/refactor some methods to reduce nesting, improve readability, and take advantage of new JavaScript features (e.g., `Array.flat`).
* Replace all references to the `getPluginActionDailog` method with usages of `useShowPluginActionDialog` or the HOC equivalent.
* Remove the old (and now, unused) `getPluginActionDailog` method.

## Testing Instructions

***TL;DR:** Regression testing for all plugin management functions.*

* In the Jetpack Cloud dashboard for agencies/partners, experiment with the various available plugin management actions:
  * Try affecting just one plugin, just one site, multiples of either, or multiples of both.
  * Try accepting the confirmation modal, and try canceling it.
  * Pay special attention to the modal's verbiage to check that the action name, site title and/or plugin name (if applicable), and spelling are all correct.
* In Calypso Blue, do much the same testing on the `/plugins/{pluginSlug}` and `/plugins/{pluginSlug}/{siteSlug}` pages. (NOTE: You'll likely need access to a WPCOM Business or eCommerce site for this step.) Confirm that all functionality works as expected.

### Reference screenshots

image

image

image

image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202978931490015

### CODE REVIEW

File: client/my-sites/plugins/hooks/types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export type Site = {
+	ID: number;
+	title: string;
+	canUpdateFiles?: boolean;
+};
+
+export type Plugin = {
+	slug: string;
+	sites: Record< string, unknown >;
```
Comment (TheSteveK) [inline client/my-sites/plugins/hooks/types.ts:23]: Just a question because I haven't seen it before.... what is this Record of which you speak?

File: client/my-sites/plugins/utils.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { isMagnificentLocale } from '@automattic/i18n-utils';
-import { useTranslate, translate } from 'i18n-calypso';
+import { useTranslate } from 'i18n-calypso';
```
Comment (TheSteveK) [inline client/my-sites/plugins/utils.js:3]: I see where you removed translate which is good.  But, it looks like the code that incorrectly used 'translate' was deleted, but I don't see any references to useTranslate elsewhere in the file.  Did I just miss it?

File: client/my-sites/plugins/hooks/types.ts (hunk 1 above)
Comment (mattgawarecki) [reply client/my-sites/plugins/hooks/types.ts:23]: Ah, yeah -- it's one of the built-in TypeScript types, basically an object where (in this case) the keys are `string`s and the values are of some unknown type:

https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type

File: client/my-sites/plugins/utils.js (hunk 2 above)
Comment (mattgawarecki) [reply client/my-sites/plugins/utils.js:3]: It's not immediately visible because it's not part of my changes, but if you unfold the unchanged parts of the file, we're using useTranslate on line 19 in `useLocalizedPlugins`.

**Review by TheSteveK — APPROVED** [review]
My bad, I forgot to approve this, it got lost in my AFK.  🚀

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8679210

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8679210).
